pkg/routeagent_driver/handlers/mtu/mtuhandler.go

```diff
@@ -123,6 +151,15 @@ func (h *mtuHandler) LocalEndpointCreated(endpoint *submV1.Endpoint) error {
 		}
 	}
 
+	if h.forceMss == needed {
+		err := h.forceMssClamping(endpoint)
```

Comment:
Any specific reason why we are programming these rules in the LocalEndpointCreated API. Can we program in the Init() method itself. i.e., Just the ruleSpecSource/ruleSpecDest will change depending on Globalnet is enabled or not.

Reply:
I also thought to add the rules in init() function, but since we want to calculate the MSS value based on the cable driver , I moved it to LocalEndpointCreated function where we have handy the cable driver.
But, yeah agree that it's more straightforward to have the rules in init() function.

Any idea how we can retrieve the cable driver in case we would like to add the rules in init() function? 

One more thing, I'm not sure how critical it is, but current PR doesn't cover runtime config change of cable driver,  I mean  something like the following use case:

SM deployed with Globalnet and VxLAN --> clamp iptables rules will be configured with value based on VxLAN header
Cable driver config changed from VxLAN to IPsec --> I don't think that mtuhandler is being called upon endpoint update

So, we'll endup with force clamp rules to based on VxLAN instead of ipsec.

Reply:
> I also thought to add the rules in init() function, but since we want to calculate the MSS value based on the cable driver , I moved it to LocalEndpointCreated function where we have handy the cable driver. But, yeah agree that it's more straightforward to have the rules in init() function.

> Any idea how we can retrieve the cable driver in case we would like to add the rules in init() function?

In that case, we can read the submariner CR object which has both the CableDriver info as well as GlobalCIDR info in the retry loop. The Submariner CR would be read by the submariner-operator while deploying Submariner components, so we can expect it to be present when the init() method is invoked.
CC: @skitt do you see any issue if we use information from Submariner CR in route-agent?


> One more thing, I'm not sure how critical it is, but current PR doesn't cover runtime config change of cable driver, I mean something like the following use case:
> 
> SM deployed with Globalnet and VxLAN --> clamp iptables rules will be configured with value based on VxLAN header Cable driver config changed from VxLAN to IPsec --> I don't think that mtuhandler is being called upon endpoint update
> 
> So, we'll endup with force clamp rules to based on VxLAN instead of ipsec.

IMO if the user is interested in updating the cableDriver, we expect the user to run "subctl uninstall" before deploying with VxLAN cable driver. As part of uninstall the rules will be flushed and fresh rules will be programmed, so it should be fine.

Reply:
> In that case, we can read the submariner CR object which has both the CableDriver info as well as GlobalCIDR info in the retry loop. The Submariner CR would be read by the submariner-operator while deploying Submariner components, so we can expect it to be present when the init() method is invoked. CC: @skitt do you see any issue if we use information from Submariner CR in route-agent?
> 

I don't think we want the route agent reading the Submariner CR. The operator already passes the GlobalCIDR via env var, why not pass the cable driver as well? Or read it from the Backend of the local Endpoint as it does now.

However, looking ahead, assuming a single, static cable driver will be further problematic for supporting multiple cable drivers, should we ever revisit that.

Reply:
In that case maybe it is worth retrieving the cable driver from endpoint CR and configuring iptables rules in LocalEndpointCreated as implemented in current PR version.

In case multiple cable drivers will be supported   I assume we'll need to configure the iptables rules in LocalEndpointCreated function 

@sridhargaddam what do you think ?